lib/jsdom/level2/events.js

```diff
@@ -267,12 +265,33 @@ core.AttributeList.prototype.$setNode = attrNodeInterceptor(core.AttributeList.p
 core.AttributeList.prototype.$set = attrInterceptor(core.AttributeList.prototype.$set, true);
 core.AttributeList.prototype.$setNoNS = attrInterceptor(core.AttributeList.prototype.$setNoNS, false);
 
-core.Document.prototype.createEvent = function(eventType) {
-    switch (eventType) {
-        case "MutationEvents": return new core.MutationEvent(eventType);
-        case "UIEvents": return new core.UIEvent(eventType);
-        case "MouseEvents": return new core.MouseEvent(eventType);
-        case "HTMLEvents": return new core.Event(eventType);
+core.Document.prototype.createEvent = function(type) {
+    var Constructor = null;
+    var typeLower = type.toLowerCase();
+    var interfaceTable = {
+      'event':            core.Event,
+      'events':           core.Event,
+      'htmlevents':       core.Event,
+      'mouseevent':       core.MouseEvent,
+      'mouseevents':      core.MouseEvent,
+      'uievent':          core.UIEvent,
+      'uievents':         core.UIEvent,
+
+      // alias unimplemented interfaces to Event
+      'customevent':      core.Event, // 'CustomEvent',
+      'keyboardevent':    core.Event, // 'KeyboardEvent',
+      'keyevents':        core.Event, // 'KeyboardEvent',
+      'messageevent':     core.Event, // 'MessageEvent',
+      'touchevent':       core.Event, // 'TouchEvent',
+
+      // old, not part of spec anymore
+      'mutationevents':   core.MutationEvent
     }
-    return new core.Event(eventType);
+
+    Constructor = interfaceTable[typeLower] || null;
+    if (!Constructor) {
+      this.raise('error', 'The provided event type (' + type + ') is invalid.', {error: 'NotSupportedError' });
```

Comment:
Message looks good. But instead of using raise, do `throw new core.DOMException(core.DOMException.NOT_SUPPORTED_ERR, ...)`